packages/namekit-react/src/components/TruncatedText.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * This TruncatedText component is a duplicate of @namehash/nameguard-react
+ * TruncatedText component. This duplicate will, in the near future, be the
+ * only TruncatedText component in the NameKit libraries. The duplicate is
+ * necessary for the moment so that it is included in main branch while
+ * @namehash/nameguard-react still has its TruncatedText: why is that?
+ *
+ * So any imports of @namehash/nameguard-react TruncatedText component
+ * do not break once this change is merged into main! Once this duplicate
+ * is in main, NameHash Labs will remove the TruncatedText component from
+ * @namehash/nameguard-react and update the imports to use this component
+ * instead, the @namehash/namekit-react TruncatedText.
+ *
+ * A ShortCut story was created to track this migration:
+ * https://app.shortcut.com/ps-web3/story/25261/next-steps-on-displayedname-and-truncatedtext
+ */
+
+import React, { useEffect, useState, useRef } from "react";
+import { Tooltip } from "./Tooltip";
+import cc from "classcat";
+
+interface TruncatedTextProps {
+  text: string;
+  maxTooltipWidth?: number;
+  maxDisplayWidth?: number;
+  textStylingClasses?: string;
+  tooltipTextStylingClasses?: string;
+  displayTooltipWhenTextOverflows?: boolean;
+}
+
+const DEFAULT_MAX_DISPLAY_WIDTH = 300;
+const DEFAULT_MAX_TOOLTIP_WIDTH = 400;
+
+export const TruncatedText = ({
+  text,
+  textStylingClasses = "",
+  tooltipTextStylingClasses = "",
+  displayTooltipWhenTextOverflows = true,
+  maxTooltipWidth = DEFAULT_MAX_TOOLTIP_WIDTH,
+  maxDisplayWidth = DEFAULT_MAX_DISPLAY_WIDTH,
```

Comment:
I would consider adding two additional props:

```ts
renderTooltip,
tooltipProps = {},
```

This makes the component more composable that allows us to pass a custom tooltip structure if necessary. Think instances of Popover card vs Tooltip.

Reply:
@notrab Thanks for feedback. Shared a related reply in another thread here: https://github.com/namehash/namekit/pull/381/files#r1733631164

Reply:
Hi!

Happy to share I have created this story to keep the progress on this thread's subject:

[TruncatedText with custom integrators Tooltip?](https://app.shortcut.com/ps-web3/story/25710/truncatedtext-with-custom-integrators-tooltip)